src/transforms/filter.js

```diff
@@ -157,18 +161,45 @@ exports.calcTransform = function(gd, trace, opts) {
     }
 };
 
-function getDataToCoordFunc(gd, trace, filtersrc) {
-    var ax = axisIds.getFromTrace(gd, trace, filtersrc);
+function getFilterArray(trace, target) {
+    if(typeof target === 'string' && target) {
+        var array = Lib.nestedProperty(trace, target).get();
+
+        return Array.isArray(array) ? array : [];
+    }
+    else if(Array.isArray(target)) return target.slice();
+
+    return false;
+}
+
+function getDataToCoordFunc(gd, trace, target) {
+    var ax;
+
+    // In the case of an array target, make a mock data array
+    // and call supplyDefaults to the data type and
+    // setup the data-to-calc method.
+    if(Array.isArray(target)) {
+        var mockGd = {
+            data: [{ x: target }],
+            layout: {}
+        };
+
+        Plots.supplyDefaults(mockGd);
```

Comment:
Calling the full `supplyDefaults` here is an overkill.

@monfera factored the autoType routine inhttps://github.com/plotly/plotly.js/pull/1021/commits/5b71b47d0ed1d437708da2a293481e911828865a, I might just use that instead once #1021 is merged.